prepDevRelease.js

```diff
@@ -0,0 +1,126 @@
+ */
+function getDevSuffixNum(devVer, latestVer) {
+  // if there is no dev version returned, make the first one
+  if (devVer === undefined) {
+    return 0;
+  }
+  // there is a dev version returned, so grab the devSuffix from it and increment.
+  const devIndex = devVer.indexOf('-dev.') + '-dev.'.length;
+  if (devIndex === -1) {
+    throw new Error(
+      `The dev tagged release \'${devVer}\'in the feed is not named properly and does not contain \'-dev\'. Please resolve this first.`,
+    );
+  }
+  const devSuffixNum = parseInt(devVer.substring(devIndex));
+  if (devSuffixNum === NaN) {
+    throw new Error(
+      `The dev tagged release \'${devVer}\'in the feed is not named properly and contains a non-number character after \'-dev.\'. Please resolve this first.`,
+    );
+  }
+
+  const newDevSuffixNum = devSuffixNum + 1;
+  const latestPrefix = latestVer;
+  const devPrefix = getPrefix(devVer);
+  const latestPatch = parseInt(latestPrefix.substring(latestPrefix.lastIndexOf('.') + 1));
+  const devPrefixPatch = parseInt(devPrefix.substring(devPrefix.lastIndexOf('.') + 1));
+  // If the current devPrefix is already higher than the latest version's prefix, there has already been a dev version
+  // released after a production version, so we'll need to just bump the dev suffix by one.
+  if (latestPatch + 1 === devPrefixPatch) {
+    return newDevSuffixNum;
+  // If the current devPrefix is the same as the latest version's prefix, it means there hasn't been a dev version
+  // released since the production release. Set the dev suffix as 0.
+  } else if (latestPrefix === devPrefix) {
+    return 0;
+  } else {
+    throw new Error(
+      `Inconsistent tags in npm feed. There shouldn't be a dev version that differs from the latest 
+      version by more than one patch version. latest version is ${latestVer} while dev version is 
+      ${devVer}. Please resolve this issue in the npm feed first.`,
+    );
+  }
+}
+
+(async () => {
+  const packageJson = getPackageJson();
+
+  // get package version from package.json
+  let currVersion = getFileVersion(packageJson);
+  console.log('package.json version: ' + currVersion);
+
+  // Find version tagged dev
+  const { devStdout, ignore } = await exec(`npm view @microsoft/teams-js version --tag dev`);
+  const newDevSuffix = getDevSuffixNum(devStdout, currVersion);
+  console.log('dev version suffix number: ' + newDevSuffix);
+
+  const [major, minor, patch] = currVersion.split('.');
+  let newDevPrefix = '';
+
+  if (devStdout !== undefined && getPrefix(devStdout) === currVersion) {
+    newDevPrefix = currVersion;
+  } else {
+    newDevPrefix = `${major}.${minor}.${parseInt(patch) + 1}`;
+  }
+
+  // append the suffix to form a new version
+  const newVersion = newDevPrefix + '-dev.' + newDevSuffix;
+
+  console.log('new version: ' + newVersion);
+
+  // update package.json with the new version
+  packageJson.version = newVersion;
+
+  savePackageJson(packageJson);
```

Comment:
Might be out of scope for this PR but it might actually make sense to have UTs for all the logic in this file to avoid regressing it accidentally when we need to evolve it in the future.

Reply:
@aeunms I think we should open a work item to track adding tests separately for this particular file. We can work on scheduling it after our current milestone.

Reply:
Created and tracked through this work item now: https://office.visualstudio.com/MetaOS/_workitems/edit/5472937